File: core/src/commonMain/kotlin/work/socialhub/kmastodon/api/NodeResource.java
package work.socialhub.kmastodon.api;

import mastodon4j.entity.Node;
import mastodon4j.entity.share.Response;

/**
 * @author uakihir0
 */
public interface NodeResource {

    /**
     * Getting node information.
     *
     * @return the node information. Does not require authentication
     */
    public Response<Node> getNodeInfo();
}

File: core/src/commonMain/kotlin/work/socialhub/kmastodon/api/NodeResource.kt
package work.socialhub.kmastodon.api

import mastodon4j.entity.Node
import mastodon4j.entity.share.Response

/**
 * @author uakihir0
 */
interface NodeResource {
    /**
     * Getting node information.
     *
     * @return the node information. Does not require authentication
     */
    val nodeInfo: Response<Node?>?
}
